docs/design/data-modeling.md

```diff
@@ -78,20 +78,31 @@ When selecting an `id` scheme:
 - An account and a transfer may share the same `id` — they belong to different "namespaces".
```

Comment:
I remember this was a discussion a few months back.

Can you remember if there was any advantage to allowing/suggesting that accounts/transfers may collide IDs?

I would rather we go the other way on this, to warn against schemes that do this, because it:

* increases surface area (we can always expand, but it's harder to restrict later),
* may be incompatible with single namespaces when exporting into other systems, and
* breaks the principle of “universally” unique (doesn't feel right).

Reply:
The reason it doesn't feel right to me is that, for example, in cryptography, there's the principle of "single use":

Only ever use something (like a symmetric key) for one purpose, one intent, and bind the use to that intent (e.g. for encryption in transit, use a HKDF to take a master symmetric key and derive 4 keys, 2 per direction, bound to the direction of traffic along the network connection). When that's violated, a CVE is typically not far off.

In our context, if a transfer is accidentally specified as an account, or vice-versa, and reinterpreted like that against the wrong data, then that could be bad. There's one less defense-in-depth, compared to if the IDs are unique.

Reply:
I completely agree that we shouldn't recommend it. My intent here was not to suggest that users deliberately collide account+transfer IDs, just to state that it is possible (because it is).

Reworded to:

> - An account and a transfer may share the same `id` (they belong to different "namespaces"),
>  but this is not recommended.

Alternatively, I can remove that bullet entirely.

Reply:
Thanks, that makes sense. We should just motivate why it's not recommended. If we say why, then it gives people a framework to evaluate the recommendation and makes them more likely to understand.

For example, "but this is not recommended because other systems (that you may later connect to TigerBeetle) may use a single "namespace" for all objects."